### ISSUE TITLE
Handling null pointer before IDE startup

### ISSUE THREAD

**Issue by SwiftExpat** [DelphiWorlds/TOTAL#5]
I found a null pointer in one of the helper functions. It happens if the method is called before IDE Startup happens. I am a little surprised this locks the IDE in a loop. It would be nice if there was an IDEStarted property to check, I could not find one.

I have a suggestion:
- add a boolean parameter, default to false,  to control the show message view, the calling code would then specify to show the message view.  The inline help would show / inform the client.

At minimum swallow this exception

What do you think is the right way to move forward?

class procedure TOTAHelper.AddTitleMessage(const AMsg: string; const AGroupName: string = '');
var
  LServices: IOTAMessageServices;
  LGroup: IOTAMessageGroup;
begin
  LServices := BorlandIDEServices as IOTAMessageServices;
  if not AGroupName.IsEmpty then
  begin
    LGroup := LServices.GetGroup(AGroupName);
    if LGroup = nil then
      LGroup := LServices.AddMessageGroup(AGroupName);
    LServices.AddTitleMessage(AMsg, LGroup);
    try
      LServices.ShowMessageView(LGroup);
    except
      on E: Exception do
      //during IDE startup this throws an exception and locks the IDE
      { TODO : Is there a Messages.ready to check before showing? }
    end;
  end
  else
    LServices.AddTitleMessage(AMsg);
end;

**Comment by DelphiWorlds** [comment]
I usually just ensure that I do not attempt anything that requires the IDE to actually be present until the IDEStarted method (which can be overridden) is called. Can you provide a scenario and/or reason why you are adding a message before then?

**Comment by SwiftExpat** [comment]
The scenario was during create of my TIDENotifierOTAWizard, I created buttons on a tool menu and one of them was sending a log message.  I set my status text, " Kill processes enabled" and down stream it is logging that to the console.  
I have modified my code to keep a boolean for IDE started, and call AddTitleMessage after IDE started.

I am split on whether to swallow the exception in a empty try except.  I can not help but wonder if there are other instances during the lifetime of the IDE where ShowMessageView would cause the loop behavior.

I lean to swallow the exception because TOTAL is a framework protecting me from the dangers of the IDE.  During a debug session the developer would still get the first chance to know something is wrong and a comment in the code would let them know not to call it before IDEStarted.
